src/main/user-impl/java/com/mysql/cj/jdbc/ha/MultiHostConnectionProxy.java

```diff
@@ -316,10 +314,8 @@ boolean isMasterConnection() {
      * @throws SQLException
      *             if an error occurs
      */
-    protected void invalidateCurrentConnection() throws SQLException {
-        synchronized (this.lockObject) {
-            invalidateConnection(this.currentConnection);
-        }
+    protected synchronized void invalidateCurrentConnection() throws SQLException {
```

Comment:
We can remove synchronized here. InvalidateConnection() has synchronized so no need to do it twice.